py/desitarget/skyhealpixs.py

```diff
@@ -0,0 +1,98 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""
+desitarget.skyhealpixs
+====================
+
+Dynamic lookup of whether a given RA,Dec location is a good place to
+put a sky fiber.
+Scripts based on desitarget.skybricks, adapted for healpix split.
+"""
+import os
+import numpy as np
+
+class Skyhealpixs(object):
+    '''
+    This class handles dynamic lookup of whether a given (RA,Dec)
+    should make a good location for a sky fiber.
+    '''
+    def __init__(self, skyhealpixs_dir=None):
+        '''
+        Create a Skyhealpixs object, reading metadata.
+
+        Parameters
+        ----------
+        skyhealpixs_dir : :class:`str`
+            The directory to find skyhealpixs data files; if None, will read from
+            SKYHEALPIXS_DIR environment variable.
+        '''
+        import fitsio
+        if skyhealpixs_dir is None:
+            skyhealpixs_dir = os.environ.get('SKYHEALPIXS_DIR', None)
+        if skyhealpixs_dir is None:
+            raise RuntimeError('Environment variable SKYHEALPIXS_DIR is not set; '
+                               'needed to look up dynamic sky fiber positions')
+        self.skyhealpixs_dir = skyhealpixs_dir
+        # AR healpix settings hard-coded...
+        self.nside, self.nest = 64, True
+
+    def lookup_position(self, ras, decs):
+        '''
+        Looks up a set of RA,Dec positions that are all within a given
+        tile (disk on the sky).
+
+        Parameters
+        ----------
+        ras : :class:`~numpy.ndarray`
+            array of RA locations to look up
+        decs : :class:`~numpy.ndarray`
+            array of Dec locations to look up
+
+        Returns
+        -------
+        :class:`~numpy.array`
+            Boolean array, same length as *ras*/*decs* inputs, of `good_sky` values.
+            (True = good place to put a sky fiber)
+        '''
+        import fitsio
+        from astropy.wcs import WCS
+        from desiutil.log import get_logger
+        import healpy as hp
+        log = get_logger()
+
+        # handle non-array iterables (eg lists) as inputs
+        ras = np.array(ras)
+        decs = np.array(decs)
+        good_sky = np.zeros(ras.shape, bool)
+        # AR healpix pixels
+        pixs = hp.ang2pix(self.nside, np.radians((90. - decs)), np.radians(ras), nest=self.nest)
```

Comment:
I'd write this as `hp.ang2pix(self.nside,ras,decs,lonlat=True,nest=self.nest)`

Reply:
Historical context: We have a request from Stephen from way back not to use the `lonlat=True` option for backwards compatibility. That request may have expired, but the standard throughout `desitarget` is _not_ to use that kwarg.

Reply:
older versions of healpy in the wild don't support lonlat keyword, so it's still helpful to avoid it when possible (we've had N>>1 dependency incompatibility issues with healpy installations, so I prefer to be as generic in our support of it's versions as possible to give people flexibility on what they can install)